lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -189,6 +189,7 @@ def add_card(post, card, action, options = {})
         post[:card][:installments_id] = options[:installments_id] if options[:installments_id]
         post[:card][:force_type] = options[:force_type].to_s.upcase if options[:force_type]
         post[:card][:save] = options[:save] if options[:save]
+        post[:card][:network_payment_reference] =  options[:stored_credential][:network_transaction_id] if options.dig(:stored_credential, :initiator) == 'merchant'
```

Comment:
Please move out all those [stored credential fields that are inside the if statement for NetworkTokenization ](https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/d_local.rb#L166) to here

Reply:
Could you also update to only send NTID if it's present